staging/src/kubevirt.io/client-go/api/v1/types.go

```diff
@@ -2003,6 +2004,14 @@ type MediatedHostDevice struct {
 	ExternalResourceProvider bool   `json:"externalResourceProvider,omitempty"`
 }
 
+// MediatedDevicesConfiguration holds inforamtion about MDEV types to be defined, if available
+// +k8s:openapi-gen=true
+type MediatedDevicesConfiguration struct {
+	// +listType=atomic
+	MediatedDevicesTypes []string            `json:"mediatedDevicesTypes,omitempty"`
+	NodeMediatedDevices  map[string][]string `json:"nodeMediatedDevices,omitempty"`
+}
+
```

Comment:
Here's a thought. Rather than a global mdev type list with per node override. What if the entire configuration was based on node selectors?

For someone who simply wants a global mdev type config across all nodes, they'd use the ```kubevirt.io/schedulable: "true"``` label for the node selector. or even the ```node-role.kubernetes.io/worker: ""``` label would work. 

For people with heterogeneous clusters that need a more expressive way of telling kubevirt what mdev types to use, then they can set multiple custom mdev type lists with corresponding node selectors to match whatever their cluster topology is.


The result could be something like this for the global mdev use case.

```
apiVersion: kubevirt.io/v1
kind: KubeVirt
metadata:
  name: kubevirt
  namespace: kubevirt
spec:
  configuration:
    mediatedDevicesConfigurations:
    - nodeSelector:
        kubevirt.io/schedulable: "true"
      mediatedDevicesTypes:
       - nvidia-222
       - nvidia-228
       - i915-GVTg_V5_4
```

And, then if you wanted different mdev types used depending on the node, you could set custom labels on your nodes and map the configuration something like this below.

```
apiVersion: kubevirt.io/v1
kind: KubeVirt
metadata:
  name: kubevirt
  namespace: kubevirt
spec:
  configuration:
    mediatedDevicesConfigurations:
    - nodeSelector:
        mdev/nvidia-222: ""
      mediatedDevicesTypes:
       - nvidia-222
    - nodeSelector:
        mdev/nvidia-228: ""
      mediatedDevicesTypes:
       - nvidia-228
```

So, a node with the label `mdev/nvidia-222` would get only the nvidia-222 mdev devices configured locally.

And a node with both the label `mdev/nvidia-222` and `mdev/nvidia-228` would get both devices configured locally (assuming there are enough cards)

Reply:
We've discussed this offline. I think that's a great idea. I didn't think about using node selectors.
I'll add this in a follow-up PR. In this PR, I'll just keep the default list of mdevs to configure and will remove the per-host override.